crates/assists/src/handlers/replace_string_with_char.rs

```diff
@@ -0,0 +1,141 @@
+use syntax::{
+    ast::{self, HasStringValue},
+    AstToken,
+    SyntaxKind::STRING,
+};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: replace_string_with_char
+//
+// Replace string with char.
+//
+// ```
+// fn main() {
+//     find("{<|>");
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     find('{');
+// }
+// ```
+pub(crate) fn replace_string_with_char(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let token = ctx.find_token_at_offset(STRING).and_then(ast::String::cast)?;
+    let value = token.value()?;
+    let target = token.syntax().text_range();
+
+    if value.is_empty() || value.chars().count() > 1 {
```

Comment:
```suggestion
    if value.chars().count() != 1 {
```

but its probably best to avoid [premature pessimization](https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#premature-pessimization) and call `.next` twice manually instead of O(N) count

other then this, lgmt!

bors d+